File: 4080-smallest-missing-multiple-of-k/4080-smallest-missing-multiple-of-k.java
class Solution {
    public int missingMultiple(int[] nums, int k) {
        HashSet<Integer> set=new HashSet<>();
        int max=0;
        for(int i:nums){
            set.add(i);
            if(i%k==0) max=Math.max(max,i);
        }
        for(int i=k;i<=max;i+=k){
            if(!set.contains(i)) return i;
        }
        if(max<k) return k; 
        return max+k;
    }
}

File: 4080-smallest-missing-multiple-of-k/4080-smallest-missing-multiple-of-k.cpp
class Solution {
public:
    int missingMultiple(vector<int>& nums, int k) {
        set<int> set;
        int m=0;
        for(int i:nums){
            set.insert(i);
            if(i%k==0) m=max(m,i);
        }
        for(int i=k;i<=m;i+=k){
            if(set.find(i) == set.end()) return i;
        }
        if(m<k) return k; 
        return m+k;
    }
};